mmv1/products/compute/RegionResizeRequest.yaml

```diff
@@ -0,0 +1,382 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: 'RegionResizeRequest'
+api_resource_type_kind: InstanceGroupManagerResizeRequest
+min_version: beta
+description: |
+  Represents a Regional Managed Instance Group Resize Request
+
+  Resize Requests are the Managed Instance Group implementation of Dynamic Workload Scheduler Flex Start.
+
+  With Dynamic Workload Scheduler in Flex Start mode, you submit a GPU capacity request for your AI/ML jobs by indicating how many you need, a duration, and your preferred region. Dynamic Workload Scheduler intelligently persists the request; once the capacity becomes available, it automatically provisions your VMs enabling your workloads to run continuously for the entire duration of the capacity allocation.
+references:
+  guides:
+    'ABOUT_RESIZE_REQUEST_IN_MIG': 'https://cloud.google.com/compute/docs/instance-groups/about-resize-requests-mig'
+  # Link to the REST API reference for the resource.
+  api: 'https://cloud.google.com/compute/docs/reference/rest/beta/regionInstanceGroupManagerResizeRequests'
+docs:
+### List Method ###
+base_url: 'projects/{{project}}/regions/{{region}}/instanceGroupManagers/{{instance_group_manager}}/resizeRequests'
+### Get Method
+self_link: 'projects/{{project}}/regions/{{region}}/instanceGroupManagers/{{instance_group_manager}}/resizeRequests/{{name}}'
+immutable: true
+timeouts:
+  insert_minutes: 20
+  update_minutes: 20
+  delete_minutes: 20
+# Sets parameters for handling operations returned by the API.
+async:
+  actions: ['create', 'delete', 'update']
+  type: 'OpAsync'
+  # Overrides which API calls return operations. Default: ['create',
+  # 'update', 'delete']
+  # actions: ['create', 'update', 'delete']
+  operation:
+    base_url: '{{op_id}}'
+
+### Update method ###
+# Resize requests are currently not update-able
+
+### Delete Method ###
+# Custom delete method to handle resize request cancellations vs. deletions.
+# If a resize request is in the CREATING or ACCEPTED state, it must be canceled before it can be
+# deleted. If a resize request is NOT in any of the mentioned state, it can be directly deleted.
+custom_code:
+  custom_delete: 'templates/terraform/custom_delete/compute_rmig_resize_request_delete.go.tmpl'
```

Comment:
I think this would probably be easier to implement / maintain if you do it as a [pre_delete](https://googlecloudplatform.github.io/magic-modules/develop/custom-code/#pre_post_injection) instead of overriding the entire delete method - if I understand correctly, you just need to call a specific method before deletion if the resource is in a specific state, and otherwise it should just need to follow the normal deletion flow. Does that seem reasonable?